interceptor/opencensus/opencensus.go

```diff
@@ -53,40 +53,38 @@ func (oci *OCInterceptor) Config(tcs agenttracepb.TraceService_ConfigServer) err
 	return errUnimplemented
 }
 
+type spansAndNode struct {
+	spans []*tracepb.Span
+	node  *commonpb.Node
+}
+
 // Export is the gRPC method that receives streamed traces from
 // OpenCensus-traceproto compatible libraries/applications.
 func (oci *OCInterceptor) Export(tes agenttracepb.TraceService_ExportServer) error {
 	// Firstly we MUST receive the node identifier to initiate
 	// the service and start accepting exported spans.
 	const maxTraceInitRetries = 15 // Arbitrary value
 
-	var node *commonpb.Node
+	var initiatingNode *commonpb.Node
 	for i := 0; i < maxTraceInitRetries; i++ {
 		recv, err := tes.Recv()
 		if err != nil {
 			return err
 		}
 
 		if nd := recv.Node; nd != nil {
-			node = nd
+			initiatingNode = nd
 			break
 		}
 	}
 
-	if node == nil {
-		return fmt.Errorf("failed to receive a non-nil Node even after %d retries", maxTraceInitRetries)
+	if initiatingNode == nil {
```

Comment:
Why is this a special case? Also if the first message contains Node and Spans we don't collect the spans.